state-chain/runtime/src/chainflip/bitcoin_block_processor.rs

```diff
@@ -56,6 +57,27 @@ impl Hook<(BlockNumber, BtcEvent), ()> for ExecuteEventHook {
 		}
 	}
 }
+impl Hook<(BlockNumber, BtcEvent<VaultDepositWitness<Runtime, BitcoinInstance>>), ()>
```

Comment:
Seems could factor some stuff out with deposit witnessing above - though idm leaving this for now until the code settles a bit